File: src/3-Problem07.pas
program problem7;

var
    row : real;
    i,j : real;

function faktorial(x : real) : real;
begin
    if (x = 0) then
    begin    
        faktorial := 1;
    end
    else
    begin
        faktorial := x*faktorial(x-1);
    end;
end;

function C(n : real; r : real) : real ;
begin
    C:= faktorial(n) / ( faktorial(r) * faktorial(n-r) );
end;

begin
    write('Masukan jumlah baris : '); readln(row);
    i := 0;
    while ( i < row ) do
    begin
        j := 0;
        while (j <= i) do
        begin
            write(C(i,j):0:0,' ');
            j := j + 1;
        end;
        i := i + 1;
        writeln;
    end;
end.

File: src/3-Problem07.cpp
#include <iostream>
using namespace std;

int C(int i, int j);
// Fungsi ini akan mengembalikan nilai iCj
// yaitu kombinasi j buah benda yang mungkin dari
// i buah benda

int Faktorial (int i);
// Fungsi ini akan mengembalikan nilai i!
// yaitu nilai ix(i-1)x(i-2)...2x1 

int main()
{
	int N;
	int i, j;

	cout << "Masukan jumlah baris: ";
	cin >> N;

	for (i = 0; i < N; ++i)
	{
		for (j = 0; j <= i; ++j)
		{
			cout << C(i,j) << " ";
		}
		
		cout << endl;
	}

	return 0;
}

int C(int i, int j) {
	return Faktorial(i)/(Faktorial(j)*(Faktorial(i-j)));
}

int Faktorial (int i) {
	int count = 1;
	int j;

	for (j = 1; j <= i; ++j)
	{
		count *= j;
	}

	return count;
}